torchaudio/transforms.py

```diff
@@ -284,22 +291,102 @@ class SpectogramToDB(object):
-    return MelSpectrogram(*args, **kwargs)
+class MFCC(object):
+    """Create the Mel-frequency cepstrum coefficients from an audio signal
+
+        Args:
+        sr (int) : sample rate of audio signal
+        n_mfcc (int) : number of mfc coefficients to retain
+        dct_type (int) : type of DCT (discrete cosine transform) to use
+        norm (string) : norm to use
+        melkwargs (dict, optional): arguments for MelSpectrogram
+    """
+    def __init__(self, sr=16000, n_mfcc=40, dct_type=2, norm='ortho', melkwargs=None):
+
+        supported_dct_types = [2]
+        if dct_type not in supported_dct_types:
+            raise ValueError('DCT type not supported'.format(dct_type))
+        self.sr = sr
+        self.n_mfcc = n_mfcc
+        self.dct_type = dct_type
+        self.norm = norm
+        self.melkwargs = melkwargs
+        self.top_db = 80.
+        self.s2db = SpectrogramToDB("power", self.top_db)
+
+        if melkwargs is not None:
+            self.MelSpectrogram = MelSpectrogram(sr=self.sr, **melkwargs)
+        else:
+            self.MelSpectrogram = MelSpectrogram(sr=self.sr)
+
+    def dct(self, x, norm=None):
+        """
+        Discrete Cosine Transform, Type II (a.k.a. the DCT)
+        For the meaning of the parameter `norm`, see:
+        https://docs.scipy.org/doc/scipy-0.14.0/reference/generated/scipy.fftpack.dct.html
+        :param x: the input signal
+        :param norm: the normalization, None or 'ortho'
+        :return: the DCT-II of the signal over the last dimension
+        """
+        x_shape = x.size()
+        l = x.new_tensor(x_shape[-1])
+
+        slices_base = [slice(x_shape_i) for x_shape_i in x_shape[:-1]]
+        v = torch.cat((x[slices_base + [slice(0, None, 2)]],
+                       x[slices_base + [slice(1, None, 2)]].flip([-1])),
+                      dim=-1)  # (*, l)
+        Vc = torch.rfft(v, 1, onesided=False)  # (*, l, 2)
+
+        k = -torch.arange(l, dtype=x.dtype, device=x.device) * np.pi / (2 * l)
+        for _ in range(Vc.dim() - 2):
+            k.unsqueeze_(0)
+        W_r, W_i = k.cos(), k.sin()  # (1,..., l)
+
+        # V is (*, l)
+        V = Vc.narrow(-1, 0, 1).squeeze(-1) * W_r - Vc.narrow(-1, 1, 1).squeeze(-1) * W_i
+
+        if norm == 'ortho':
+            V_ndim_0 = V.narrow(-1, 0, 1)
+            V_ndim_rest = V.narrow(-1, 1, x_shape[-1] - 1)
+
+            V_ndim_0.div_(torch.sqrt(l) * 2)
+            V_ndim_rest.div_(torch.sqrt(l / 2) * 2)
+
+        V.mul_(2.)
+
+        return V
+
+    def __call__(self, sig):
+        """
+        Args:
+            sig (Tensor): Tensor of audio of size (channels [c], samples [n])
+
+        Returns:
+            spec_mel_db (Tensor): channels x hops x n_mels (c, l, m), where channels
+                is unchanged, hops is the number of hops, and n_mels is the
+                number of mel bins.
+
+        """
+        mel_spect_db = self.s2db(self.MelSpectrogram(sig))
```

Comment:
This is dangerous. Any text book or wikipedia will tell that MFCCs are computed by a DCT on **logarithmic** mel spectrograms. This should be clearly documented, and possibly made an option.
(To be fair, I just checked librosa, and they also compute MFCCs from db-scaled mel spectrograms. I'm really surprised.)

Reply:
Yes I was surprised by that too. I think we could add a parameter for log vs. db-scale and not complicate things too much. What do you think?